crates/sdk/src/network.rs

```diff
@@ -0,0 +1,185 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use sp1_core_machine::io::SP1Stdin;
+use sp1_prover::{SP1ProvingKey, SP1VerifyingKey};
+use std::future::Future;
+use std::pin::Pin;
+
+use crate::{NetworkClient, ProveAction, SP1VerificationError};
+
+pub struct NetworkProver {
+    local_prover: CpuProver,
+    network_client: NetworkClient,
+}
+
+impl NetworkProver {
+    pub fn new(rpc_url: String, private_key: String) -> Self {
+        Self {
+            local_prover: CpuProver::new(),
+            network_client: NetworkClient::new(&private_key).with_rpc_url(rpc_url),
+        }
+    }
+
+    /// Get the cycle limit to used for a proof request.
+    ///
+    /// The cycle limit is determined according to the following priority:
+    /// 1. If a cycle limit was explicitly set, use the specified value
+    /// 2. If simulation is enabled (default), calculate the limit by simulating
+    /// 3. Otherwise use the default cycle limit
+    #[allow(clippy::must_use_candidate)]
+    fn get_cycle_limit(
+        &self,
+        elf: &[u8],
+        stdin: &SP1Stdin,
+        cycle_limit: Option<u64>,
+        skip_simulation: bool,
+    ) -> Result<u64, Error> {
+        todo!()
+    }
+
+    /// Registers a program if it is not already registered.
+    pub async fn register_program(
+        &self,
+        vk: &SP1VerifyingKey,
+        elf: &[u8],
+    ) -> Result<VerifyingKeyHash> {
+        todo!()
+    }
+
+    /// Requests a proof from the prover network, returning the request ID.
+    #[allow(clippy::too_many_arguments)]
+    pub async fn request_proof(
+        &self,
+        vk_hash: &VerifyingKeyHash,
+        stdin: &SP1Stdin,
+        version: &str,
+        mode: ProofMode,
+        strategy: FulfillmentStrategy,
+        timeout_secs: u64,
+        cycle_limit: u64,
+    ) -> Result<RequestId> {
+        todo!()
+    }
+
+    /// Waits for the proof request to be fulfilled by the prover network.
+    ///
+    /// The proof request must have already been submitted. This function will return a
+    /// `RequestTimedOut` error if the request does not received a response within the timeout.
+    pub async fn wait_proof<P: DeserializeOwned>(
+        &self,
+        request_id: &RequestId,
+        timeout_secs: u64,
+    ) -> Result<P, Error> {
+        todo!()
+    }
+
+    pub fn prove_with_options(
```

Comment:
I think `prove_with_options` could just be reduced to `prove` or `prove_with_request` is better

Reply:
I agree I think that returning a `ProofRequest` with option methods replaces this